tests/unit/local/docker/test_container.py

```diff
@@ -481,19 +481,40 @@ def setUp(self):
 
         self.container.is_created = Mock()
 
-    def test_must_start_container(self):
+    @patch("socket.socket")
+    def test_must_start_container(self, patched_socket):
 
         self.container.is_created.return_value = True
 
         container_mock = Mock()
         self.mock_docker_client.containers.get.return_value = container_mock
         container_mock.start = Mock()
 
+        socket_mock = Mock()
+        socket_mock.connect_ex.return_value = 0
+        patched_socket.return_value = socket_mock
+
         self.container.start()
 
         self.mock_docker_client.containers.get.assert_called_with(self.container.id)
         container_mock.start.assert_called_with()
 
+    @patch("socket.socket")
+    def test_times_out_if_port_not_open(self, patched_socket):
```

Comment:
the test also has to wait 10 seconds (or whatever the timeout is set to) for it to time out. Is this okay?

Reply:
if you can not override them in unit testing, so it is Ok